File: src/main/java/io/vproxy/vfx/animation/AnimationEdge.java
package io.vproxy.vfx.animation;

import io.vproxy.vfx.util.algebradata.AlgebraData;
import io.vproxy.vfx.util.algebradata.DoubleDoubleFunction;
import io.vproxy.commons.graph.GraphEdge;

public class AnimationEdge<T extends AlgebraData<T>> extends GraphEdge<AnimationNode<T>> {
    public final AnimationNode<T> from;
    public final AnimationNode<T> to;
    public final long durationMillis;
    public final DoubleDoubleFunction function;

    AnimationEdge(AnimationNode<T> from, AnimationNode<T> to, long durationMillis, DoubleDoubleFunction function) {
        super(from, to, durationMillis);
        this.from = from;
        this.to = to;
        this.durationMillis = durationMillis;
        this.function = function;
    }
}

File: src-ts/src/animation/AnimationEdge.ts
import type { AlgebraData } from '../util/algebradata/AlgebraData.js';
import { GraphEdge } from '../commons-graph/Graph.js';
import type { AnimationNode } from './AnimationNode.js';
import type { DoubleDoubleFunction } from './DoubleDoubleFunction.js';

export class AnimationEdge<T extends AlgebraData<T>> extends GraphEdge<AnimationNode<T>> {
  override readonly from: AnimationNode<T>;
  override readonly to: AnimationNode<T>;
  readonly durationMillis: number;
  readonly function: DoubleDoubleFunction;

  constructor(
    from: AnimationNode<T>,
    to: AnimationNode<T>,
    durationMillis: number,
    fn: DoubleDoubleFunction,
  ) {
    super(from, to, durationMillis);
    this.from = from;
    this.to = to;
    this.durationMillis = durationMillis;
    this.function = fn;
  }
}
